10.epilepsy_path_ranking.ipynb

```diff
@@ -0,0 +1,449 @@
+    "\n",
+    "metagraph = hetio.readwrite.read_metagraph(url)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_paths_for_metapath(metagraph, row):\n",
+    "    '''\n",
+    "    Return a list of dictionaries containing the information for all paths with a given source, target, and metapath\n",
+    "    \n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    metagraph : a hetio.hetnet.Metagraph instance to interpret metapath abbreviations\n",
+    "    row : a row from a pandas dataframe with information about the given metapath, source, and target\n",
+    "    '''\n",
+    "    damping_exponent = .5\n",
+    "    \n",
+    "    metapath_data = metagraph.metapath_from_abbrev(row['metapath_id'])\n",
+    "\n",
+    "    query = hetio.neo4j.construct_pdp_query(metapath_data, path_style='string', property='name')\n",
+    "\n",
+    "    driver = GraphDatabase.driver(\"bolt://neo4j.het.io\")\n",
+    "    params = {\n",
+    "        'source': row['source_name'],\n",
+    "        'target': row['target_name'],\n",
+    "        'w': damping_exponent\n",
+    "    }\n",
+    "    with driver.session() as session:\n",
+    "        metapath_result = session.run(query, params)\n",
+    "        metapath_result = metapath_result.data()\n",
+    "\n",
+    "    for path in metapath_result:\n",
+    "        path['metapath'] = row['metapath_id']\n",
+    "        path['metapath_importance'] = row['neg_log_p_value']\n",
+    "        path['path_importance'] = path['metapath_importance'] * path['percent_of_DWPC']\n",
+    "        path['source'] = row['source_name']\n",
+    "    \n",
+    "    metapath_df = pandas.DataFrame(metapath_result)\n",
+    "        \n",
+    "    return metapath_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "425300bf31a04eedbf36fc8b94b430f2",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "HBox(children=(IntProgress(value=0, max=6740), HTML(value='')))"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# For row in top_metapaths\n",
+    "\n",
+    "result_df = None\n",
+    "\n",
+    "for index, row in tqdm.tqdm_notebook(top_metapaths.iterrows(), total=len(top_metapaths.index)):\n",
+    "    metapath_df = get_paths_for_metapath(metagraph, row)\n",
+    "    if result_df is None:\n",
+    "        result_df = metapath_df\n",
+    "    else:\n",
+    "        result_df = pandas.concat([metapath_df, result_df], ignore_index=True)"
```

Comment:
greater efficiency probably to append each df to a list and then concat the list once having exited loop.

If it'd be cumbersome to rerun. Just save this advice for next time.